oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/incrementalstore/MergeIncrementalFlatFileStore.java

```diff
@@ -0,0 +1,220 @@
+                    if (compared < 0) { // write baseFFSLine in merged file and advance line in baseFFS
+                        baseFFSLine = writeAndAdvance(writer, baseFFSBufferedReader, baseFFSLine);
+                    } else if (compared > 0) { // write incrementalFFSline and advance line in incrementalFFS
+                        String[] incrementalFFSParts = IncrementalFlatFileStoreNodeStateEntryWriter.getParts(incrementalFFSLine);
+                        checkState(IncrementalStoreOperand.ADD.toString().equals(getOperand(incrementalFFSParts)));
+                        incrementalFFSLine = writeAndAdvance(writer, incrementalFFSBufferedReader,
+                                getFFSLineFromIncrementalFFSParts(incrementalFFSParts));
+                    } else {
+                        String[] incrementalFFSParts = IncrementalFlatFileStoreNodeStateEntryWriter.getParts(incrementalFFSLine);
+                        String operand = getOperand(incrementalFFSParts);
+                        switch (enumMap.get(operand)) {
+                            case MODIFY:
+                                incrementalFFSLine = writeAndAdvance(writer, incrementalFFSBufferedReader,
+                                        getFFSLineFromIncrementalFFSParts(incrementalFFSParts));
+                                break;
+                            case DELETE:
+                                incrementalFFSLine = incrementalFFSBufferedReader.readLine();
+                                break;
+                            default:
+                                log.error("wrong operand in incremental ffs: operand:{}, line:{}", operand, incrementalFFSLine);
+                                throw new RuntimeException("wrong operand in incremental ffs: operand:" + operand + ", line:" + incrementalFFSLine);
+                        }
+                        baseFFSLine = baseFFSBufferedReader.readLine();
+                    }
+                } else {
+                    if (incrementalFFSLine == null) {
+                        baseFFSLine = writeRestOfFFSFileAndAdvance(writer, baseFFSBufferedReader, baseFFSLine);
+                    } else {
+                        incrementalFFSLine = writeRestOfIncrementalFileAndAdvance(writer, incrementalFFSBufferedReader, incrementalFFSLine);
+                    }
+                }
+            }
+        }
+    }
+
+    private IndexStoreMetadata getIndexStoreMetadataForMergedFile() throws IOException {
+        File baseFFSMetadataFile = IndexStoreUtils.getMetadataFile(baseFFS, algorithm);
+        File incrementalMetadataFile = IndexStoreUtils.getMetadataFile(incrementalFFS, algorithm);
+
+        if (baseFFSMetadataFile.exists() && incrementalMetadataFile.exists()) {
+            IndexStoreMetadata indexStoreMetadata = new IndexStoreMetadataOperatorImpl<IndexStoreMetadata>()
+                    .getIndexStoreMetadata(baseFFSMetadataFile, algorithm, new TypeReference<IndexStoreMetadata>() {
```

Comment:
the explicit type can be removed
```suggestion
                    .getIndexStoreMetadata(baseFFSMetadataFile, algorithm, new TypeReference<>() {
```